app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,59 @@
+class VendorsController < ApplicationController
+
+  def index
+    @vendors = Vendor.uniq.order(name: :asc)
+  end
+
+  def show
+    @vendors = Vendor.where(id: params[:id])
+  end
+
+  def new
+    @vendor = Vendor.new 
+    if @vendor.market_id
+      @current_market = @vendor.market.id
+    end
+  end
```

Comment:
When you are in The market view page and want to create a new vendor for that market (you have the option to do that on your view), you probably want to save the market id and assign it to the vendor. 
I would add a check   if !params[market_id].nil?